apm-agent-core/src/main/java/co/elastic/apm/agent/bci/ElasticApmAgent.java

```diff
@@ -683,10 +683,10 @@ public static void ensureInstrumented(Class<?> classToInstrument, Collection<Cla
                         .with(FailSafeDeclaredMethodsCompiler.INSTANCE);
                     AgentBuilder agentBuilder = getAgentBuilder(byteBuddy, config, logger, AgentBuilder.DescriptionStrategy.Default.HYBRID, false);
                     for (Class<? extends ElasticApmInstrumentation> instrumentationClass : instrumentationClasses) {
+                        ElasticApmInstrumentation apmInstrumentation = instantiate(instrumentationClass);
                         pluginClassLoaderByAdviceClass.put(
-                            instrumentationClass.getName(),
+                            apmInstrumentation.getAdviceClass().getName(),
```

Comment:
without this, it works only with advice that is directly within the instrumentation class, and not delegated to it's own class.

Reply:
Can we do this caching based on advice class names only, instead of instrumentation classes and their **instantiation** here? I mean that `ensureInstrumented` will be done directly on advice class names